bundles/framework/rpc/service/RpcService.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * @class Oskari.mapframework.bundle.rpc.service.RpcService
+ * Allows extend RPC supported funtions
+ *
+ */
+Oskari.clazz.define('Oskari.mapframework.bundle.rpc.service.RpcService',
+
+    /**
+     * @method create called automatically on construction
+     * @static
+     *
+     */
+
+    function (instance) {
+        this.instance = instance;
+        this.conf = this.instance.conf || {};
+        this.channel = this.instance._channel;
+        this._log = Oskari.log('RpcService');
+        this._availableFunctions = {};
+        this._allowedFunctions = this.conf.allowedfunctions || [];
+        this._allowedFunctionsFromConf = Array.isArray(this.conf.allowedfunctions);
+    }, {
+        __name: 'Rpc.RpcService',
+        __qname: 'Oskari.mapframework.bundle.rpc.service.RpcService',
+
+        getQName: function () {
+            return this.__qname;
+        },
+
+        getName: function () {
+            return this.__name;
+        },
+
+        /**
+         * @method init
+         * Initializes the service
+         */
+        init: function () {
+        },
+
+        /**
+         * @method addFunction
+         * Add function to RPC
+         *
+         * @param func function that should be added
+         */
+        addFunction: function (func) {
+            if (!func && typeof func !== 'function') {
+                return;
+            }
+
```

Comment:
Check that the function has a name and is not called with `addFunction(() => { /* TODO impl */})` or `addFunction(function () { /* TODO impl */ });`
```suggestion

            if (!func.name || !func.name.trim()) {
                me._log.warn('Trying add anonymous RPC function, skipping. Provide name for function.');
                return;
            }
```

Reply:
Then again processors and transpilers might change the name of a function or remove it. Should we pass the name as string with additional param to this function?

Reply:
added name param